tests/behaviour/steps/params.rs

```diff
@@ -65,11 +65,11 @@ impl MayError {
 
     pub fn check_concept_write_without_read_errors<T: fmt::Debug>(
         &self,
-        res: &Result<T, ConceptWriteError>,
-    ) -> Option<ConceptWriteError> {
+        res: &Result<T, Box<ConceptWriteError>>,
+    ) -> Option<Box<ConceptWriteError>> {
         match self.check(res.as_ref().map_err(|e| e.clone())) {
             None => None,
-            Some(error) => match error {
+            Some(error) => match *error {
                 ConceptWriteError::ConceptRead { source } => {
                     panic!("Expected logic error, got ConceptRead {:?}", source)
                 }
```

Comment:
I'd prefer to see something more like this:
```rs
    pub fn check_concept_write_without_read_errors<'a, T: fmt::Debug, E: AsRef<ConceptWriteError>>(
        &self,
        res: &'a Result<T, E>,
    ) -> Option<&'a E> {
        match self.check(res.as_ref().map_err(E::as_ref)) {
            None => None,
            Some(error) => match error {
                ConceptWriteError::ConceptRead { source } => {
                    panic!("Expected logic error, got ConceptRead {:?}", source)
                }
```
So that you don't have to box random errors just for this check.
Also, do we need ownership of the return here at all? I assumed not.

Reply:
I tried this and reverted it since i couldn't make it work with one of the call sites - also feel free to tweak it tmrw after i merge